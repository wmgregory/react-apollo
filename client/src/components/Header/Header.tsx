import React from 'react';
import styled from 'styled-components';

interface HeaderProps {
  title: string;
}

const StyledHeader = styled('header')`
  display: flex;
  flex-direction: column;

  color: white;
  background-color: #282c34;
`;

const StyledHeading = styled('h1')`
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

export const Header: React.FC<HeaderProps> = ({ title }) => {
  return (
    <StyledHeader>
      <StyledHeading>{title}</StyledHeading>
    </StyledHeader>
  );
};
