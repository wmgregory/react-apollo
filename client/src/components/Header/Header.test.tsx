import React from 'react';

import { shallow } from 'enzyme';

import { Header } from './Header';

describe('Header', () => {
  it('should render heading', () => {
    const wrapper = shallow(<Header title='New Title' />);

    expect(wrapper).toBeDefined();
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
