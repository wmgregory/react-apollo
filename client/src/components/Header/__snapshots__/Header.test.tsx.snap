// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render heading 1`] = `
"<header className=\\"App-header\\">
  <h1>
    New Title
  </h1>
</header>"
`;
